<launch>
  <arg name="controller_config_file" default="$(find z1_ros_control)/config/controllers.yaml" />
  <arg name="sdk_config_file" default="$(find z1_ros_control)/config/config.yaml" />
  <arg name="controllers" default="joint_state_controller z1_joint_traj_controller" />
  <arg name="UnitreeGripperYN" default="true"/>

  <rosparam file="$(arg controller_config_file)" command="load"/>

  <node name="ros_control_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" args="$(arg controllers)" />

  <node type="z1_ros_control_node" pkg="z1_ros_control" name="z1_ros_control_node" output="screen">
    <param name="UnitreeGripper" value="$(arg UnitreeGripperYN)"/>

    <rosparam file="$(arg sdk_config_file)" command="load"/>
  </node>
</launch>